flink/v1.13/flink/src/main/java/org/apache/iceberg/flink/FlinkSchemaUtil.java

```diff
@@ -149,14 +156,43 @@ public static TableSchema toSchema(RowType rowType) {
    * @param schema iceberg schema to convert.
    * @return Flink TableSchema.
    */
-  public static TableSchema toSchema(Schema schema) {
+  public static TableSchema toSchema(Schema schema, Map<String, String> properties) {
     TableSchema.Builder builder = TableSchema.builder();
 
     // Add columns.
     for (RowType.RowField field : convert(schema).getFields()) {
       builder.field(field.getName(), TypeConversions.fromLogicalToDataType(field.getType()));
     }
 
+    // Add watermark
+    final int watermarkCount =
+        properties.keySet().stream()
+            .filter(
+                (k) ->
```

Comment:
should we use Java regex matcher for more accurate match?

Reply:
This logic is similar to flink's. So i think it is more suitable, and we can update it with flink version in future.

Reply:
ok. sounds good to me